dotnet/src/SemanticKernel.Core/Data/ServiceCollectionExtensions.cs

```diff
@@ -21,6 +22,44 @@ public static class ServiceCollectionExtensions
     public static IServiceCollection AddVolatileVectorStore(this IServiceCollection services, string? serviceId = default)
     {
         services.AddKeyedSingleton<IVectorStore, VolatileVectorStore>(serviceId);
+        services.AddKeyedSingleton<VolatileVectorStore, VolatileVectorStore>(serviceId);
```

Comment:
I believe we are registering two separate volatile vector store instances here.  Consider something like this to only have one.
```csharp
services.AddKeyedSingleton<VolatileVectorStore, VolatileVectorStore>(serviceId);
services.AddKeyedSingleton<IVectorStore>(serviceId, sp => sp.GetRequiredKeyedService<VolatileVectorStore>(serviceId));
```